cli/module_loader.rs

```diff
@@ -544,16 +547,15 @@ impl CliModuleLoader {
     &self,
     referrer: &str,
   ) -> Result<ModuleSpecifier, AnyError> {
-    // TODO(bartlomieju): ideally we shouldn't need to call `current_dir()` on each
-    // call - maybe it should be caller's responsibility to pass it as an arg?
-    let cwd = std::env::current_dir().context("Unable to get CWD")?;
     if referrer.is_empty() && self.shared.is_repl {
       // FIXME(bartlomieju): this is a hacky way to provide compatibility with REPL
       // and `Deno.core.evalContext` API. Ideally we should always have a referrer filled
       // but sadly that's not the case due to missing APIs in V8.
+      let cwd = std::env::current_dir().context("Unable to get CWD")?;
       deno_core::resolve_path("./$deno$repl.ts", &cwd).map_err(|e| e.into())
     } else {
-      deno_core::resolve_url_or_path(referrer, &cwd).map_err(|e| e.into())
+      deno_core::resolve_url_or_path(referrer, &self.shared.initial_cwd)
+        .map_err(|e| e.into())
```

Comment:
> I believe this might break Deno.core.evalContext which CommonJS module system relies on. Any chance you could add a test for it where you change the CWD then call Deno.core.evalContext and dynamically import a file with relative path?

@bartlomieju do you have a real world example with cjs where this would fail?

Reply:
Maybe what we should only look up the cwd when the referrer is empty?

Reply:
Yeah, that's what I meant with my comment - for `Deno.core.evalContext` the referrer will be empty. Maybe it's not the case for CJS per se, but raw `Deno.core.evalContext` might break. I'm not 100% sure, just something that came to mind reviewing it.